dashboard/app/models/ai_tutor_interaction.rb

```diff
@@ -0,0 +1,31 @@
+# == Schema Information
+#
+# Table name: ai_tutor_interactions
+#
+#  id                 :bigint           not null, primary key
+#  user_id            :integer          not null
+#  level_id           :integer
+#  script_id          :integer
+#  ai_model_version   :string(255)
+#  type               :string(255)
+#  project_id         :string(255)
+#  project_version_id :string(255)
+#  prompt             :text(65535)
+#  status             :string(255)
+#  ai_response        :text(65535)
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+# Indexes
+#
+#  index_ai_tutor_interactions_on_level_id   (level_id)
+#  index_ai_tutor_interactions_on_script_id  (script_id)
+#  index_ai_tutor_interactions_on_user_id    (user_id)
+#  index_ati_user_level_script               (user_id,level_id,script_id)
+#
+class AiTutorInteraction < ApplicationRecord
+  belongs_to :user
+  belongs_to :level, optional: true
+  belongs_to :script, optional: true
+  validates :type, inclusion: {in: %w(compilation validation general_chat)}
```

Comment:
I keep these [validation examples](https://dev.to/daviducolo/rails-model-validation-a-comprehensive-guide-with-code-examples-21mh) bookmarked, and the [rails docs](https://guides.rubyonrails.org/active_record_validations.html) are good too if you want more details especially about when the validations are checked.